logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/message/ephemeral/EphemeralMessageDeletionHandlerTest.kt

```diff
@@ -0,0 +1,413 @@
+package com.wire.kalium.logic.feature.message.ephemeral
+
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.message.Message
+import com.wire.kalium.logic.data.message.MessageRepository
+import com.wire.kalium.logic.framework.TestMessage
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.test_util.TestKaliumDispatcher
+import io.mockative.Mock
+import io.mockative.Times
+import io.mockative.any
+import io.mockative.classOf
+import io.mockative.given
+import io.mockative.mock
+import io.mockative.verify
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.advanceTimeBy
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.seconds
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class EphemeralMessageDeletionHandlerTest {
+
+    private val testDispatcher = TestKaliumDispatcher
+
+    @Test
+    fun givenRegularMessage_whenEnqueueingForFirstTime_thenSelfDeletionShouldBeMarked() = runTest(
+        testDispatcher.default
+    ) {
+        // given
+        val (arrangement, ephemeralMessageDeletionHandler) = Arrangement(
+            coroutineScope = this,
+            dispatcher = testDispatcher
+        ).withMessageRepositoryReturningMessage(
+            message = TestMessage.TEXT_MESSAGE.copy(
+                expirationData = Message.ExpirationData(
+                    expireAfter = 1.seconds,
+                    selfDeletionStatus = Message.ExpirationData.SelfDeletionStatus.NotStarted
+                )
+            )
+        ).withMessageRepositoryMarkingSelfDeletionStartDate().arrange()
+
+        // when
+        ephemeralMessageDeletionHandler.startSelfDeletion(
+            conversationId = ConversationId("someValue", "someDomain"),
```

Comment:
I am not sure I follow ? We are not matching anywhere conversationId and messageId to return a specific message, so I do not think it makes any difference

Reply:
In case we change `ConversationId` and remove/add parameters, we will need to come here and fix this place.

It's a test and the values don't matter for this one.

We could just use `TestConversation.ID` or similar.

If it breaks, we can fix in that one place.

Reply:
I am not I follow, what you guys are trying to say here :( could u explain me the issue ?

Reply:
I meant that the message you are specifying to be returned when querying the `messageRepository` above, should have the same `conversationId` that you are providing here. That's why I mentioned these test variables should be defined/setup at the beginning of the test. 
That way we maximise the readability of the test and we can also verify more accurately that the calls to the mocked `messageRepository` were made with the same id parameters.

Reply:
Making a test verification with `any()` is a way of fast tracking a test, but also leaving some space for things to be broken silently.

Reply:
The fact that the test passes doesn't mean it is correctly written. So even if this test doesn't break anything atm, it may be not be robust enough to catch an error in the production code related with this action.

For example, if we call on the production code to:
`ephemeralMessageDeletionHandler.startSelfDeletion(ConversationId(value1, domain1), messageId1)` 

and then for any reason, there is an error on the `messageRepository` and mistakenly (newly added logic error, issue with the intermediate message mapper class, etc...) we make a wrong call in the lower layer like:

`messageRepository.markSelfDeletionStartDate(ConversationId(value1, domain2), messageId1)`

This test would still pass, but clearly, the message marked for deletion is not the correct one.

Reply:
That's why I said that we should first declare some immutable test variables at the beginning of the test, then use their same `conversationId` and `messageId` parameters to invoke the method.
And lastly make sure on the mocking stubs, that they are invoked with: 
```
.with(matching { it.value == expectedConversation.value && it.domain == expectedConversation.domain}, eq(expectedMessageId))
```
That way we can ensure that the correct message is being used for both invoking the handler method and the lower message repository layers.

Reply:
Thanks for clarifying it on 1 on 1 it indeed makes sense